### PR TITLE
Server unit tests implementation & coding convention update.

### PR DESCRIPTION
**Pull request by nvthongswansea**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -125,250 +90,230 @@ type ServerCreateRequestIsoimage struct {
...
+func (c *Client) GetServerMetricList(id string) ([]ServerMetric, error) {
 	r := Request{
-		uri:    apiServerBase + "/" + id + "/power",
-		method: "PATCH",
-		body:   body,
+		uri:    path.Join(apiServerBase, id, "metrics"),
+		method: http.MethodGet,
+	}
+	var response ServerMetricList
+	var list []ServerMetric
```
Comment (itakouna) [inline server.go]: replace `list` with a proper name.

File: server.go [hunk 2]
```diff
@@ -125,250 +90,230 @@ type ServerCreateRequestIsoimage struct {
...
-	return c.WaitForServerPowerStatus(id, false)
+	return server.Properties.Power, nil
 }
 
-func (c *Client) ShutdownServer(id string) error {
-	//Make sure the server exists and that it isn't already in the state we need it to be
-	server, err := c.GetServer(id)
+//turnOnOffServer turn on/off a specific server.
+//turnOn=true to turn on, turnOn=false to turn off
+func (c *Client) turnOnOffServer(id string, turnOn bool) error {
```
Comment (itakouna) [inline server.go]: I would suggest `changeServerPowerState` instead of `turnOnOffServer`

File: serverip.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	r := Request{
+		uri:    path.Join(apiServerBase, id, "ips"),
+		method: http.MethodPost,
+		body:   body,
+	}
+	return r.execute(*c, nil)
+}
+
+//DeleteServerId delete a link between a storage and an IP
+func (c *Client) DeleteServerId(serverId, ipId string) error {
```
Comment (itakouna) [inline serverip.go]: `ipId` Let's follow the rules https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: serverip.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	r := Request{
+		uri:    path.Join(apiServerBase, serverId, "ips", ipId),
+		method: http.MethodDelete,
+	}
+	return r.execute(*c, nil)
+}
+
+//LinkIp attaches an IP to a server
+func (c *Client) LinkIp(serverid string, ipid string) error {
```
Comment (itakouna) [inline serverip.go]: `ipid` same here

File: serverip.go [hunk 5]
```diff
@@ -59,24 +58,24 @@ func (c *Client) CreateServerIp(id string, body ServerIpCreateRequest) error {
 	return r.execute(*c, nil)
 }
 
-//DeleteServerId delete a link between a storage and an IP
-func (c *Client) DeleteServerId(serverId, ipId string) error {
+//DeleteServerIp delete a link between a storage and an IP
```
Comment (itakouna) [inline serverip.go]: `storage` should not be `server`?

Comment (nvthongswansea) [reply serverip.go]: Sorry, my bad.

**Pull request merged**